core/src/solution_submission.rs

```diff
@@ -388,42 +465,8 @@ mod tests {
         let mut contract = MockStableXContract::new();
         contract
             .expect_submit_solution()
-            .times(1)
-            .with(always(), always(), always(), eq(U256::from(5)), eq(Some(2)))
-            .return_once(|_, _, _, _, _| {
-                async {
-                    Err(MethodError::from_parts(
-                    "submitSolution(uint32,uint256,address[],uint16[],uint128[],uint128[],uint16[])"
-                        .to_owned(),
-                    ExecutionError::ConfirmTimeout,
-                ))
-                }
-                .boxed()
-            });
-        contract
-            .expect_submit_solution()
-            .times(1)
-            .with(
-                always(),
-                always(),
-                always(),
-                eq(U256::from(12)),
-                eq(Some(2)),
-            )
-            .return_once(|_, _, _, _, _| {
-                async {
-                    Err(MethodError::from_parts(
-                    "submitSolution(uint32,uint256,address[],uint16[],uint128[],uint128[],uint16[])"
-                        .to_owned(),
-                    ExecutionError::ConfirmTimeout,
-                ))
-                }
-                .boxed()
-            });
-        contract
-            .expect_submit_solution()
-            .with(always(), always(), always(), eq(U256::from(15)), eq(None))
-            .return_once(|_, _, _, _, _| {
+            .times(3)
+            .returning(|_, _, _, _, _| {
```

Comment:
Since the gas price calculation is independently tested we don't need this test to test the exact values.